spec/p2p/v0.34/pex-protocol.md

```diff
@@ -0,0 +1,240 @@
+# Peer Exchange Protocol
+
+The Peer Exchange (PEX) protocol enables nodes to exchange peer addresses, thus
+implementing a peer discovery mechanism.
+
+The PEX protocol uses two messages:
+
+- `PexRequest`: sent by a node to [request](#requesting-addresses) peer
+  addresses to a peer
+- `PexAddrs`: a list of peer addresses [provided](#providing-addresses) to a
+  peer as response to a `PexRequest` message
+
+While all nodes, with few exceptions, participate on the PEX protocol,
+a subset of nodes, configured as [seed nodes](#seed-nodes) have a particular
+role in the protocol.
+They crawl the network, connecting to random peers, in order to learn as many
+peer addresses as possible to provide to other nodes.
+
+## Requesting Addresses
+
+A node requests peer addresses by sending a `PexRequest` message to a peer.
+
+For regular nodes, not operating in seed mode, a PEX request is sent when
+the node *needs* peers addresses, a condition checked:
+
+1. When an *outbound* peer is added, causing the node to request addresses from
+   the new peer
+1. Periodically, by the `ensurePeersRoutine`, causing the node to request peer
```

Comment:
`1.` -> `2.`

Reply:
Thank you for taking the time to review the spec. We have merged this PR because we are preparing a next one with new edits and will then open a PR against `main`. I will incorporated your suggestions there though.